lib/buffer.js

```diff
@@ -170,18 +170,31 @@ Object.defineProperty(Buffer, Symbol.species, {
  * Buffer.from(arrayBuffer[, byteOffset[, length]])
  **/
 Buffer.from = function(value, encodingOrOffset, length) {
+  if (value === undefined || value === null)
+    throw new TypeError(kFromErrorMsg);
+
+  if (typeof value === 'number')
+    throw new TypeError('"value" argument must not be a number');
+
   if (typeof value === 'string')
```

Comment:
yeah, good point, I updated and shifted the number check back down a bit. It cannot be at the bottom like before because doing so causes the valueOf option to go into a recursion loop.